confdgnmi/src/confd_gnmi_server.py

```diff
@@ -57,6 +56,52 @@ def get_and_connect_adapter(self, username, password):
         log.debug("<== adapter=%s", adapter)
         return adapter
 
+    def get_val_encoding(self, val):
+        log.debug("==> val=%s", val)
+        encoding = None
+        if str(val).startswith("json_val"):
+            encoding = gnmi_pb2.Encoding.JSON
+        elif str(val).startswith("json_ietf_val"):
+            encoding = gnmi_pb2.Encoding.JSON_IETF
+        elif str(val).startswith("ascii_val"):
+            encoding = gnmi_pb2.Encoding.ASCII
+        elif  str(val).startswith("bytes_val"):
+            encoding = gnmi_pb2.Encoding.BYTES
+        else:
+            found = False
+            for a in ["string_val", "int_val", "uint_val", "bool_val", "bytes_val", "float_val", "leaflist_val"]:
+                if  str(val).startswith(a):
+                    found = True
+                    break
+            if found:
+                encoding = gnmi_pb2.Encoding.PROTO
+        log.debug("<== encoding=%s", encoding)
+        return encoding
+
+    def _encoding_supported(self, encoding, adapter, context):
```

Comment:
- name is unclear -> does not hint on what this method does
- seems to be incorrectly placed as object method? it does not use "self" anywhere...
  (or some of it's arguments are unnecessary and could be retrieved from self instead?)